### PR TITLE
fix(ci): Install hadolint in pre-commit workflow

### PR DESCRIPTION
**Pull request by Techassi**
Part of https://github.com/stackabletech/issues/issues/637

This fixes two things:

- Install the hadolint binary to be able to run the pre-commit hook
- Set the default language version of `node` to `system`. This fixes issues on NixOS systems.

### CODE REVIEW

File: .github/workflows/pr_pre-commit.yaml [hunk 1]
```diff
@@ -14,6 +17,17 @@ jobs:
       - uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
         with:
           python-version: '3.12'
+      - name: Setup Hadolint
+        shell: bash
+        run: |
+          set -xeuo pipefail
+
+          LOCATION="$HOME/hadolint"
```
Comment (NickLarsenNZ) [inline .github/workflows/pr_pre-commit.yaml:25]: Does it need adding to the GITHUB_PATH?

**Review by NickLarsenNZ — COMMENTED** [review]
I'm not sure $HOME is in the path.

File: .github/workflows/pr_pre-commit.yaml (hunk 1 above)
Comment (Techassi) [reply .github/workflows/pr_pre-commit.yaml:25]: Yeah that's the thing I'm not sure about. I didn't find any information on that in their documentation.

Comment (Techassi) [reply .github/workflows/pr_pre-commit.yaml]: Indeed, `$HOME` didn't work, we need to use `$GITHUB_PATH`.

File: .github/workflows/pr_pre-commit.yaml [hunk 2]
```diff
@@ -14,6 +17,22 @@ jobs:
       - uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
         with:
           python-version: '3.12'
+      - name: Setup Hadolint
+        shell: bash
+        run: |
+          set -xeuo pipefail
```
Comment (NickLarsenNZ) [inline .github/workflows/pr_pre-commit.yaml:23]: We can remove debug now, if you want. I don't mind if it stays in though.

```suggestion
          set -euo pipefail
```

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Pull request merged**